addons/web/static/src/legacy/scss/utils.scss

```diff
@@ -124,6 +124,31 @@
     }
 }
 
+// Some elements default to the text color of the body. However, there are cases
+// where the body text color is adjusted to contrast the background color of the
+// body, causing issues for elements with a different background color. This
+// function adjusts the text color of such elements to ensure visibility based
+// on their own background.
+@function adjust-body-text-color-for-background($background-color) {
+    // Checks if the body text color is light.
+    @if color-yiq($body-color) == $yiq-text-dark {
+        @return color-yiq($background-color, $body-color, $yiq-text-dark);
+    } @else {
+        @return color-yiq($background-color, $yiq-text-light, $body-color);
+    }
+}
+
+// We do the same thing as in the “adjust-body-text-color-for-background”
+// function but for the color of muted texts.
+@function adjust-muted-text-color-for-background($background-color) {
```

Comment:
Instead of creating two functions that do the same thing, make it only one function with two parameters. Name the function and parameters that way:
```scss
@function adjust-color-to-background($color, $background-color) {
    ...
```

Reply:
> Instead of creating two functions that do the same thing, make it only one function with two parameters.

I created two separate functions because I use the `mute-color()` function for muted colors.
But indeed, I can create a single function with 3 parameters (lightColor, darkColor, background).

Reply:
> I created two separate functions because I use the mute-color() function for muted colors.

Indeed, I forgot about that

> But indeed, I can create a single function with 3 parameters (lightColor, darkColor, background).

I don't think you can, that would just be the `color-yiq` function :smile: What you can do however is adding two optional parameter to the function I mentioned:
```scss
@function adjust-color-to-background($color, $background-color, $light: $yiq-text-light, $dark: $yiq-text-dark) {
```

That way for the body color stuff you can call it with:
```scss
color: adjust-color-to-background($body-color, <your_background>)
```

And for the text-muted stuff, you can call it with:
```scss
color: adjust-color-to-background($text-muted, <your_background>, $light: mute-color($yiq-text-light), $dark: mute-color($yiq-text-dark))
```